Bryn Mawr Classical Review.js

```diff
@@ -3,93 +3,135 @@
-		var data = doc.evaluate('//h3[i]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent;
-		var title = title.replace("(", "\\(").replace(")", "\\)");
-		var author = doc.evaluate('//b[contains(text(), "Reviewed")]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent.match(/Reviewed by\s+([^,]+),/)[1];
-		item.creators.push(Zotero.Utilities.cleanAuthor(author, "author"));
-		var splitRe = new RegExp(title);
-		var authors = data.split(splitRe)[0].replace(/\([^)]+\)/, "").split(/(,|and)\s+/);
-		Zotero.debug(authors);
-		for (var i=0; i<authors.length; i++) {
-			var aut = authors[i];
-			if (aut.match(/\w/) && (aut != "and")) {
-				item.creators.push(Zotero.Utilities.cleanAuthor(aut, "reviewedAuthor"));
-			}
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("journalArticle");
+	
+	var title = ZU.xpathText(doc, '//h3/i');
+	item.title = "Review of: " + Zotero.Utilities.trimInternal(title);
+	var title = title.replace("(", "\\(").replace(")", "\\)");
+	
+	var author = ZU.xpathText(doc, '//b[contains(text(), "Reviewed by")]');
+	if (author) {
+		author = author.match(/Reviewed by\s+([^,\(]+)/);
+		if (author) {
+			item.creators.push(ZU.cleanAuthor(author[1], "author"));
 		}
-		item.url = doc.location.href;
-		item.attachments = [{url:item.url, title:item.title, mimeType:"text/html"}];
-		if (doc.evaluate('/html/body/center/table/tbody/tr/td/center/table/tbody/tr/td//font', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext()) {
-			item.date = Zotero.Utilities.trimInternal(doc.evaluate('/html/body/center/table/tbody/tr/td/center/table/tbody/tr/td//font', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent.replace("Bryn Mawr Classical Review ", "").replace(/\./g, "/"));
-		} else {
-			item.date = Zotero.Utilities.trimInternal(doc.evaluate('/html/body/h3', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext().textContent.replace("Bryn Mawr Classical Review ", "").replace(/\./g, "/"))
+	}
+
+	//The authors of the reviewed book are also child nodes of h3
+	//and before the book title which is set in italics.
+	var dataChildrens = ZU.xpath(doc, '//h3[i]')[0].childNodes;
+	var authorString = "";
+	for (var i=0; i<dataChildrens.length; i++) {
+		if (dataChildrens[i].tagName == "I") {
+			break;
 		}
-		item.complete();
-	});
-}/** BEGIN TEST CASES **/
+		authorString += dataChildrens[i].textContent;
+	}
+	var authors = authorString.replace(/\([^)]+\)/, "").split(/(,|and)\s+/);
+	//Zotero.debug(authors);
+	for (var i=0; i<authors.length; i++) {
+		var aut = authors[i];
+		if (aut.match(/\w/) && (aut !== "and")) {
+			item.creators.push(ZU.cleanAuthor(aut, "reviewedAuthor"));
+		}
+	}
+
+	//The BMCR ID contains the 4-digit year, 2 digit month and a increasing number
+	var m = url.match(/(\d{4})-(\d{2})-(\d{2})/);//this should work for 1999f
+	if (m) {
+		item.extra = "BMCR ID: " + m[1] + "." + m[2] + "." + m[3];
+		item.date = m[1] + "-" + m[2];
+	}
```

Comment:
This should actually work from 1998 on. For years between 1994 and 1997 inclusively, the match is

`\d{2})\.(\d{2})\.(\d{2})/);`

and then we've got:

`item.date = "19" + m[1] + "-" + m[2];`